src/istore.c

```diff
@@ -167,6 +167,39 @@ Datum istore_max_key(PG_FUNCTION_ARGS)
     }
 }
 
+/*
+ * get the biggest value from an istore
+ */
+PG_FUNCTION_INFO_V1(istore_max_value);
+Datum istore_max_value(PG_FUNCTION_ARGS)
+{
+    IStore *    is;
+    IStorePair *pairs;
+    int         index;
+    int32       value;
+
+    is    = PG_GETARG_IS(0);
+    if (is->len == 0)
+    {
+        PG_RETURN_NULL();
+    }
+
+    index = 0;
+    pairs = FIRST_PAIR(is, IStorePair);
+    value = pairs[index].val;
```

Comment:
Just little optimization. You probably don't need to check first element again in the loop and just use `for` loop:

```suggestion
    pairs = FIRST_PAIR(is, IStorePair);
    value = pairs[0].val;
    for (int i = 1; i < is->len; i++)
```

Reply:
Done